tests/suite/reconfig_test.go

```diff
@@ -0,0 +1,614 @@
+
+		time.Sleep(60 * time.Second)
+
+		ns := core.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "reconfig",
+			},
+		}
+		Expect(resourceManager.Apply([]client.Object{&ns})).To(Succeed())
+		Expect(resourceManager.ApplyFromFiles(
+			[]string{
+				"reconfig/certificate-ns-and-cafe-secret.yaml",
+				"reconfig/reference-grant.yaml",
+				"reconfig/gateway.yaml",
+			},
+			ns.Name)).To(Succeed())
+
+		Expect(createUniqueResources(resourceCount, "manifests/reconfig/cafe-routes.yaml")).To(Succeed())
+
+		return nil
+	}
+
+	checkResourceCreation := func(resourceCount int) error {
+		ctx, cancel := context.WithTimeout(context.Background(), timeoutConfig.GetTimeout)
+		defer cancel()
+
+		var namespaces core.NamespaceList
+		if err := k8sClient.List(ctx, &namespaces); err != nil {
+			return fmt.Errorf("error getting namespaces: %w", err)
+		}
+		Expect(len(namespaces.Items)).To(BeNumerically(">=", resourceCount))
+
+		var routes v1.HTTPRouteList
+		if err := k8sClient.List(ctx, &routes); err != nil {
+			return fmt.Errorf("error getting HTTPRoutes: %w", err)
+		}
+		Expect(len(routes.Items)).To(BeNumerically(">=", resourceCount*3))
+
+		var pods core.PodList
+		if err := k8sClient.List(ctx, &pods); err != nil {
+			return fmt.Errorf("error getting Pods: %w", err)
+		}
+		Expect(len(pods.Items)).To(BeNumerically(">=", resourceCount*2))
+
+		return nil
+	}
+
+	cleanupResources := func(resourceCount int) {
+		for i := 1; i <= resourceCount; i++ {
+			nsName := "namespace" + strconv.Itoa(i)
+			Expect(resourceManager.DeleteNamespace(nsName)).To(Succeed())
+		}
+
+		ns := core.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "reconfig",
+			},
+		}
+
+		Expect(resourceManager.DeleteFromFiles([]string{
+			"reconfig/certificate-ns-and-cafe-secret.yaml",
+			"reconfig/reference-grant.yaml",
+			"reconfig/gateway.yaml",
+		}, ns.Name)).To(Succeed())
+	}
+
+	getTimeStampFromLogLine := func(logLine string) string {
+		var timeStamp string
+
+		timeStamp = strings.Split(logLine, "\"ts\":\"")[1]
+		// sometimes the log message will contain information on a "logger" followed by the "msg"
+		// while other times the "logger" will be omitted
+		timeStamp = strings.Split(timeStamp, "\",\"msg\"")[0]
+		timeStamp = strings.Split(timeStamp, "\",\"logger\"")[0]
+
+		return timeStamp
+	}
+
+	calculateTimeDifferenceBetweenLogLines := func(firstLine, secondLine string) (int, error) {
+		firstTS := getTimeStampFromLogLine(firstLine)
+		secondTS := getTimeStampFromLogLine(secondLine)
+
+		// i might be able to just use the local constant timestamp layout
+		layout := "2006-01-02T15:04:05Z"
+
+		parsedTS1, err := time.Parse(layout, firstTS)
+		if err != nil {
+			return 0, err
+		}
+
+		parsedTS2, err := time.Parse(layout, secondTS)
+		if err != nil {
+			return 0, err
+		}
+
+		return int(parsedTS2.Sub(parsedTS1).Seconds()), nil
+	}
+
+	calculateTimeToReadyAverage := func(ngfLogs string) (string, error) {
```

Comment:
Would like some eyes on this function.

Reply:
I always worry when automated tests depend on specific log statements because logs can change. 

It's great you got this working, and I'm sure it took some time, but I wonder if we need to capture this metric 😬 

Between the event batch processing time and the reload time, I think we will be able to catch any performance changes. If we do think that TTR is an important metric, I think the better way to collect it is by adding another metric to the code so we don't have to parse logs. 

@sjberman thoughts?

Reply:
Yeah that's a good point, logs can be risky. I do think TTR is important, but I'm trying to figure out how we can really measure it. We would need two metrics, start and end, and some way to link them together (probably using a label for the resource that was created). Though I wonder if that will cause our metrics to get out of hand when a ton of resources exist.

Reply:
Any decision we can make?

Reply:
Maybe we leave this as-is for now. We know it could be an issue, and now that we plan on running NFR tests on a regular cadence, if an issue arises we should know quickly and can address it.

Reply:
I'm fine leaving it as-is, but can we file an issue to look for a better long-term solution for collecting TTR numbers?